clients/client-python/build.gradle.kts

```diff
@@ -62,17 +64,134 @@ fun waitForServerIsReady(host: String = "http://localhost", port: Int = 8090, ti
+    if (operation == "start") {
+      waitForServerIsReady()
+    }
+    println("Gravitino server status: $currentContext")
+  } else {
+    println("Gravitino server execution failed with exit code $exitCode")
+  }
+}
+
+fun startHiveContainer() {
+  val launchScript = "${project.rootDir.path}/clients/client-python/tests/integration/init/launch.sh"
+  val processBuilder = ProcessBuilder("bash", "-c", launchScript)
+  val logPath = System.getenv("gravitino.log.path")
+  if (logPath != null && logPath.isNotBlank()) {
+    val env = mapOf("GRAVITINO_LOG_PATH" to logPath)
+    processBuilder.environment().putAll(env)
+  }
+  val process = processBuilder.start()
+  val exitCode = process.waitFor()
+  if (exitCode != 0) {
+    val output = process.inputStream.bufferedReader().readText()
+    throw RuntimeException("Hive container started failed with exit code $exitCode, msg: $output")
+  }
+}
+
+fun resolveDockerAddress(): Map<String, String> {
+  val resolveAddressScript = "${project.rootDir.path}/clients/client-python/tests/integration/init/inspect_ip.sh"
+  val process = ProcessBuilder("bash", "-c", resolveAddressScript).start()
+
+  val exitCode = process.waitFor()
+  val output = process.inputStream.bufferedReader().readText()
+  if (exitCode == 0) {
+    val lines = output.lines()
+    // expect the output to be like:
+    // hive:10.20.30.19
+    val addressMap = mutableMapOf<String, String>()
+    lines.stream()
+       .filter { line -> line.isNotBlank() }
+       .forEach { line ->
+          val (name, ip) = line.split(":").let { (name, ip) -> Pair(name, ip) }
+          if (name == "hive") {
+            if (ip.isBlank()) {
+              throw RuntimeException("Hive container address is blank.")
+            }
+            addressMap["hive"] = ip
+          }
       }
-      println("Gravitino server status: $currentContext")
-    } else {
-      println("Gravitino server execution failed with exit code $exitCode")
+    return addressMap
+  }
+  throw RuntimeException("Docker container address resolved failed with exit code $exitCode, msg: $output")
+}
+
+fun appendServerHadoopConf(hiveContainerAddress: String) {
+  val hadoopConfPath = "${project.rootDir.path}/distribution/package/catalogs/hadoop/conf/hadoop.conf"
+  val confFile = File(hadoopConfPath)
+  if (!confFile.exists()) {
+    throw RuntimeException("Hadoop conf file is not found at `$hadoopConfPath`.")
+  }
+  val confs = mapOf(
+    "gravitino.bypass.fs.defaultFS" to "hdfs://$hiveContainerAddress:9000"
+  )
+
+  BufferedWriter(FileWriter(confFile, true)).use { writer ->
+    confs.forEach { (key, value) ->
+      writer.append("\n$key = $value")
     }
+  }
+}
+
+val hadoopVersion = "2.7.3"
+val hadoopPackName = "hadoop-${hadoopVersion}.tar.gz"
+val hadoopDownloadUrl = "https://archive.apache.org/dist/hadoop/core/hadoop-${hadoopVersion}/${hadoopPackName}"
+val localArchiveDir = "${project.rootDir.path}/clients/client-python/it-archive"
+fun getAndUnzipHadoopPack() {
```

Comment:
Why not just use another shell script here to run download and unzip? 
I think writing these logics into scripts is easier to read. 
Also, because we also need to support MacOS, I think maybe it is great to use different scripts to handle different architectures (Linux, MacOS, etc.)
WDYT?

Reply:
OK, I think this makes sense. I'll refactor the logic here.